cmd/lncli/invoicesrpc_active.go

```diff
@@ -212,12 +217,14 @@ func addHoldInvoice(ctx *cli.Context) error {
 
 	args = args.Tail()
 
-	switch {
-	case ctx.IsSet("amt"):
+	if ctx.IsSet("amt") {
 		amt = ctx.Int64("amt")
-	case args.Present():
+	}
+	if ctx.IsSet("amt_msat") {
+		amtMsat = ctx.Int64("amt_msat")
+	}
+	if amt == 0 && amtMsat == 0 && args.Present() {
```

Comment:
Any reason these checks were added? To override the amount used in the flag rather than the argument?

Reply:
In old code, there was a switch, taking --amt amount if it's present, and parsing an argument otherwise.
I decided to retain compatibility, making --amt, --amt_ms "prioritized", however, some edge cases will still behave differently (--amt 0, arg 182). To attain perfect compatibility, if !ctx.IsSet("amt") && !ctx.IsSet("amt_ms") should have been used.